webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -160,6 +166,46 @@ function ChannelSettingsAccessRulesTab({
         setAutoSyncMembers((prev) => !prev);
     }, [parentPolicyAutoSync, expression, autoSyncMembers]);
 
+    // Validate that current user satisfies the expression
+    const validateSelfExclusion = useCallback(async (testExpression: string): Promise<boolean> => {
```

Comment:
We should defer this logic to backend (but we can still keep basic checks like empty ID, empty expressions etc.)